util/abs/provision_pe_xl_nodes.rb

```diff
@@ -1,9 +1,83 @@
 # frozen_string_literal: true
 
+# TODO: make this a Bolt task once abs_helper is available as a gem
+
+require "optparse"
 require "yaml"
+require "json"
 require "./setup/helpers/abs_helper.rb"
 include AbsHelper # rubocop:disable Style/MixinUsage
 
+DESCRIPTION = <<~DESCRIPTION
+  This script was created to assist in working with the pe_xl module (https://github.com/reidmv/reidmv-pe_xl).
+  It provisions the nodes used by the module and generates the Bolt inventory and parameter files populated with the provisioned hosts.
+
+  * Note: Because the script is designed to work with GPLT it sets up the environment without HA by default.
+
+  EC2 hosts are provisioned for the following roles:
+  * Core roles:
+   - master
+   - puppet_db
+   - compiler_a
+   - compiler_b
+
+  * HA roles:
+   - master_replica
+   - puppet_db_replica
+
```

Comment:
Need to describe the defaults here.

Reply:
Updated to output the defaults after the options since it seems awkward to list default values for the options without first describing them.